lms/djangoapps/learner_recommendations/utils.py

```diff
@@ -224,3 +223,83 @@ def filter_recommended_courses(
             filtered_recommended_courses.append(course_data)
 
     return filtered_recommended_courses
+
+
+def get_programs_based_on_course(course_key, country_code, user):
+    """
+    Returns a program for the course. If a course is part of multiple programs,
+    this function returns the program with the highest price.
+    """
+    # Get the programs list for the course
+    programs = get_programs(course=course_key)
+
+    # check user is enrolled in different courses part of program
+    if programs:
+        enrollable_programs = []
+        for program in programs:
+            course_keys = []
+            for courses in program.get('courses'):
+                course_keys.append(courses.get('key'))
+            enrollable_course_keys = _remove_user_enrolled_course_keys(user, course_keys)
+            if enrollable_course_keys:
+                enrollable_programs.append(program)
+
+        non_restricted_programs = []
+        for program in enrollable_programs:
+            # Check for location restriction
+            if not _has_country_restrictions(program, country_code):
+                non_restricted_programs.append(program)
+
+        max_price_program = max(non_restricted_programs, key=lambda x: x['price_ranges'][0]['total'])
+
+        total_weeks_to_complete = 0
+        course_pacing_type = ''
+
+        for courses in max_price_program.get('courses'):
+            for course_run in courses.get('course_runs'):
+                if course_run.get('availability') in ['Current', 'Upcoming']:
+                    course_pacing_type = course_run.get("pacing_type")
+                    total_weeks_to_complete += int(course_run.get("weeks_to_complete"))
+
+        get_program_duration(total_weeks_to_complete)
+        program_upsell = {
+            "program_title": max_price_program.get('title'),
+            "marketing_url": max_price_program.get('marketing_url'),
+            "total_courses": len(max_price_program.get('courses')),
+            "pacing_type": course_pacing_type,
+            "weeks_to_complete": get_program_duration(total_weeks_to_complete),
+            "weeks_to_complete_min": max_price_program.get('weeks_to_complete_min'),
+            "weeks_to_complete_max": max_price_program.get('weeks_to_complete_max'),
+            "type": max_price_program.get('type'),
+        }
+        return program_upsell
+    else:
+        return None
+
+
+def get_program_duration(weeks):
```

Comment:
This should be a private function. Also move this to where other private functions are added (at the top)